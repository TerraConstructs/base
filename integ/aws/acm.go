package aws

import (
	"fmt"
	"time"

	"github.com/aws/aws-sdk-go/service/acm"
	"github.com/stretchr/testify/require"

	"github.com/gruntwork-io/terratest/modules/aws"
	"github.com/gruntwork-io/terratest/modules/logger"
	"github.com/gruntwork-io/terratest/modules/retry"
	"github.com/gruntwork-io/terratest/modules/testing"
)

// ref: https://github.com/gruntwork-io/terratest/blob/v0.47.1/modules/aws/acm.go

// Get Certificate Status
func GetAcmCertificateStatus(t testing.TestingT, awsRegion string, certArn string) string {
	status, err := GetAcmCertificateStatusE(t, awsRegion, certArn)
	if err != nil {
		t.Fatal(err)
	}
	return status
}

// GetAcmCertificateStatusE gets the ACM certificate status for the given certificate ARN in the given region.
func GetAcmCertificateStatusE(t testing.TestingT, awsRegion string, certArn string) (string, error) {
	acmClient, err := aws.NewAcmClientE(t, awsRegion)
	if err != nil {
		return "", err
	}

	result, err := acmClient.DescribeCertificate(&acm.DescribeCertificateInput{
		CertificateArn: &certArn,
	})
	if err != nil {
		return "", err
	}

	return *result.Certificate.Status, nil
}

// WaitForCertificateIssued waits for the certificate to be issued
func WaitForCertificateIssued(
	t testing.TestingT,
	certArn string,
	region string,
	maxRetries int,
	sleepBetweenRetries time.Duration,
) {
	err := WaitForCertificateIssuedE(t, certArn, region, maxRetries, sleepBetweenRetries)
	require.NoError(t, err)
}

// WaitForCertificateIssuedE waits for the ACM Certificate to be issued
func WaitForCertificateIssuedE(
	t testing.TestingT,
	certArn string,
	region string,
	maxRetries int,
	sleepBetweenRetries time.Duration,
) error {
	msg, err := retry.DoWithRetryE(
		t,
		fmt.Sprintf("Waiting for Certificate %s to be ISSUED.", certArn),
		maxRetries,
		sleepBetweenRetries,
		func() (string, error) {
			certStatus, err := GetAcmCertificateStatusE(t, region, certArn)
			if err != nil {
				return "", err
			}
			if certStatus != acm.CertificateStatusIssued {
				return "", NewCertificateNotIssuedError(certArn, certStatus)
			}
			return fmt.Sprintf("Certificate %s is now at desired status %s", certArn, acm.CertificateStatusIssued), nil
		},
	)
	logger.Log(t, msg)
	return err
}
