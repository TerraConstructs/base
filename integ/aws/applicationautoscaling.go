package aws

import (
	"context"
	"fmt"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/applicationautoscaling"
	"github.com/aws/aws-sdk-go-v2/service/applicationautoscaling/types"
	"github.com/stretchr/testify/require"

	terratestaws "github.com/gruntwork-io/terratest/modules/aws"
	"github.com/gruntwork-io/terratest/modules/testing"
)

// GetTableTrackingPolicy gets the target tracking policy for a DynamoDB table or errors if not found
func GetTableTrackingPolicy(t testing.TestingT, awsRegion string, resourceId string) *types.ScalingPolicy {
	policy, err := GetTableTrackingPolicyE(t, awsRegion, resourceId)
	require.NoError(t, err)
	return policy
}

// GetTableTrackingPolicy gets the target tracking policy for a DynamoDB table or returns an error if not found
func GetTableTrackingPolicyE(t testing.TestingT, awsRegion string, resourceId string) (*types.ScalingPolicy, error) {
	policies := GetScalingPolicies(t, awsRegion, "dynamodb")

	var targetTrackingPolicy *types.ScalingPolicy
	for _, policy := range policies {
		if *policy.ResourceId == resourceId &&
			policy.ScalableDimension == types.ScalableDimensionDynamoDBTableReadCapacityUnits &&
			policy.PolicyType == types.PolicyTypeTargetTrackingScaling {
			targetTrackingPolicy = &policy
			break
		}
	}
	if targetTrackingPolicy == nil {
		return nil, fmt.Errorf("no target tracking policy found for resource ID: %s", resourceId)
	}
	return targetTrackingPolicy, nil
}

// GetScalableTargets gets the Application Auto Scaling scalable targets for the given service namespace
func GetScalableTargets(t testing.TestingT, region string, serviceNamespace string) []types.ScalableTarget {
	targets, err := GetScalableTargetsE(t, region, serviceNamespace)
	require.NoError(t, err)
	return targets
}

// GetScalableTargetsE gets the Application Auto Scaling scalable targets for the given service namespace
func GetScalableTargetsE(t testing.TestingT, region string, serviceNamespace string) ([]types.ScalableTarget, error) {
	client, err := NewApplicationAutoScalingClientE(t, region)
	if err != nil {
		return nil, err
	}

	result, err := client.DescribeScalableTargets(context.Background(), &applicationautoscaling.DescribeScalableTargetsInput{
		ServiceNamespace: types.ServiceNamespace(serviceNamespace),
	})

	if err != nil {
		return nil, err
	}

	return result.ScalableTargets, nil
}

// GetScalingPolicies gets the Application Auto Scaling scaling policies for the given service namespace
func GetScalingPolicies(t testing.TestingT, region string, serviceNamespace string) []types.ScalingPolicy {
	policies, err := GetScalingPoliciesE(t, region, serviceNamespace)
	require.NoError(t, err)
	return policies
}

// GetScalingPoliciesE gets the Application Auto Scaling scaling policies for the given service namespace
func GetScalingPoliciesE(t testing.TestingT, region string, serviceNamespace string) ([]types.ScalingPolicy, error) {
	client, err := NewApplicationAutoScalingClientE(t, region)
	if err != nil {
		return nil, err
	}

	result, err := client.DescribeScalingPolicies(context.Background(), &applicationautoscaling.DescribeScalingPoliciesInput{
		ServiceNamespace: types.ServiceNamespace(serviceNamespace),
	})

	if err != nil {
		return nil, err
	}

	return result.ScalingPolicies, nil
}

// GetScheduledActions gets the Application Auto Scaling scheduled actions for the given service namespace
func GetScheduledActions(t testing.TestingT, region string, serviceNamespace string) []types.ScheduledAction {
	actions, err := GetScheduledActionsE(t, region, serviceNamespace)
	require.NoError(t, err)
	return actions
}

// GetScheduledActionsE gets the Application Auto Scaling scheduled actions for the given service namespace
func GetScheduledActionsE(t testing.TestingT, region string, serviceNamespace string) ([]types.ScheduledAction, error) {
	client, err := NewApplicationAutoScalingClientE(t, region)
	if err != nil {
		return nil, err
	}

	result, err := client.DescribeScheduledActions(context.Background(), &applicationautoscaling.DescribeScheduledActionsInput{
		ServiceNamespace: types.ServiceNamespace(serviceNamespace),
	})

	if err != nil {
		return nil, err
	}

	return result.ScheduledActions, nil
}

// GetScalableTargetsByResourceId gets scalable targets filtered by resource ID
func GetScalableTargetsByResourceId(t testing.TestingT, region string, serviceNamespace string, resourceId string) []types.ScalableTarget {
	targets, err := GetScalableTargetsByResourceIdE(t, region, serviceNamespace, resourceId)
	require.NoError(t, err)
	return targets
}

// GetScalableTargetsByResourceIdE gets scalable targets filtered by resource ID
func GetScalableTargetsByResourceIdE(t testing.TestingT, region string, serviceNamespace string, resourceId string) ([]types.ScalableTarget, error) {
	client, err := NewApplicationAutoScalingClientE(t, region)
	if err != nil {
		return nil, err
	}

	result, err := client.DescribeScalableTargets(context.Background(), &applicationautoscaling.DescribeScalableTargetsInput{
		ServiceNamespace: types.ServiceNamespace(serviceNamespace),
		ResourceIds:      []string{resourceId},
	})

	if err != nil {
		return nil, err
	}

	return result.ScalableTargets, nil
}

// GetScheduledActionsByResourceId gets scheduled actions filtered by resource ID
func GetScheduledActionsByResourceId(t testing.TestingT, region string, serviceNamespace string, resourceId string) []types.ScheduledAction {
	actions, err := GetScheduledActionsByResourceIdE(t, region, serviceNamespace, resourceId)
	require.NoError(t, err)
	return actions
}

// GetScheduledActionsByResourceIdE gets scheduled actions filtered by resource ID
func GetScheduledActionsByResourceIdE(t testing.TestingT, region string, serviceNamespace string, resourceId string) ([]types.ScheduledAction, error) {
	client, err := NewApplicationAutoScalingClientE(t, region)
	if err != nil {
		return nil, err
	}

	result, err := client.DescribeScheduledActions(context.Background(), &applicationautoscaling.DescribeScheduledActionsInput{
		ServiceNamespace: types.ServiceNamespace(serviceNamespace),
		ResourceId:       aws.String(resourceId),
	})

	if err != nil {
		return nil, err
	}

	return result.ScheduledActions, nil
}

// NewApplicationAutoScalingClient creates a new Application Auto Scaling client
func NewApplicationAutoScalingClient(t testing.TestingT, region string) *applicationautoscaling.Client {
	client, err := NewApplicationAutoScalingClientE(t, region)
	require.NoError(t, err)
	return client
}

// NewApplicationAutoScalingClientE creates a new Application Auto Scaling client
func NewApplicationAutoScalingClientE(t testing.TestingT, region string) (*applicationautoscaling.Client, error) {
	sess, err := terratestaws.NewAuthenticatedSession(region)
	if err != nil {
		return nil, err
	}
	return applicationautoscaling.NewFromConfig(*sess), nil
}
