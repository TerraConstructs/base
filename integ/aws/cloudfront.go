package aws

import (
	"context"
	"encoding/json"
	"fmt"
	"strconv"

	"github.com/aws/aws-sdk-go-v2/config"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/cloudfront"
	"github.com/aws/aws-sdk-go-v2/service/cloudfront/types"
	"github.com/gruntwork-io/terratest/modules/logger"
	"github.com/gruntwork-io/terratest/modules/testing"
	"github.com/hashicorp/go-multierror"
	"github.com/stretchr/testify/require"
)

type CloudFrontTestFunctionResult struct {
	Utilization  int // The amount of time that the function took to run as a percentage of the maximum allowed time.
	ErrorMessage *string
	// The event object returned by the function. For more information about the
	// structure of the event object, see [Event object structure]in the Amazon CloudFront Developer Guide.
	//
	// [Event object structure]: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/functions-event-structure.html
	Output map[string]interface{}
	// Contains the log lines that the function wrote (if any) when running the test.
	ExecutionLogs []string
}

const invalidFunctionErrorPrefix = "The CloudFront function associated with the CloudFront distribution is invalid or could not run. Error: "

// responseValidator is a function that validates the response of a CloudFront function test.
type responseValidator func(*CloudFrontTestFunctionResult) error

// Tests a CloudFront function.
//
// To test a function, you provide an event object that represents an HTTP request
// or response that your CloudFront distribution could receive in production.
// CloudFront runs the function, passing it the event object that you provided, and
// returns the function's result (the modified event object) in the response. The
// response also contains function logs and error messages, if any exist. For more
// information about testing functions, see [Testing functions]in the Amazon CloudFront Developer
// Guide.
//
// To test a function, you provide the function's name and stage (DEVELOPMENT or LIVE)
// along with the event object. To get the function's name and version, you can use
// ListFunctions and DescribeFunction.
//
// [Testing functions]: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/managing-functions.html#test-function
func TestCloudFrontFunction(t testing.TestingT, name string, stage string, event CloudFrontFunctionEvent, expectedError string, expectedOutput *map[string]interface{}) error {
	validateResponse := responseValidator(func(r *CloudFrontTestFunctionResult) error {
		var combinedErr error
		gotError := aws.ToString(r.ErrorMessage)
		if gotError != expectedError {
			combinedErr = multierror.Append(combinedErr, fmt.Errorf("got Error :%s\nWant Error:%s", gotError, expectedError))
		}

		if expectedOutput != nil {
			expectedOutputBytes, _ := json.MarshalIndent(r.Output, "", "  ")
			if r.Output == nil {
				combinedErr = multierror.Append(combinedErr, fmt.Errorf("got nil Output\nWant Output:%v", string(expectedOutputBytes)))
			} else {
				prettyPrint, err := PrettyPrintBeforeAfter(r.Output, *expectedOutput)
				if err != nil {
					combinedErr = multierror.Append(combinedErr, fmt.Errorf("error matching expected Output:%v", err))
				}
				if prettyPrint != "" {
					combinedErr = multierror.Append(combinedErr, fmt.Errorf("expected Output mismatch (-got, +want):%s", prettyPrint))
				}
			}
		} else {
			if r.Output != nil {
				outputBytes, _ := json.MarshalIndent(r.Output, "", "  ")
				combinedErr = multierror.Append(combinedErr, fmt.Errorf("got Output:%v\nWant nil Output", string(outputBytes)))
			}
		}
		return combinedErr
	})
	return TestCloudFrontFunctionWithCustomValidation(t, name, stage, event, validateResponse)
}

// TestCloudFrontFunctionWithCustomValidation performs a Function test and validate the response. Fails the test if there is an error.
func TestCloudFrontFunctionWithCustomValidation(t testing.TestingT, name string, stage string, event CloudFrontFunctionEvent, validateResponse responseValidator) error {
	functionStage := assertFunctionStage(t, stage)
	err := TestCloudFrontFunctionWithCustomValidationE(t, name, functionStage, event, validateResponse)
	require.NoError(t, err)
	return nil
}

// TestCloudFrontFunctionWithCustomValidationE performs a Function test and validate the response.
func TestCloudFrontFunctionWithCustomValidationE(t testing.TestingT, name string, stage types.FunctionStage, event CloudFrontFunctionEvent, validateResponse responseValidator) error {
	response, err := TestCloudFrontFunctionE(t, name, stage, event)
	if err != nil {
		return err
	}
	// log utilization for information purposes
	logger.Log(t, fmt.Sprintf("CloudFront Function Utilization: (%d%%)", response.Utilization))
	if err := validateResponse(response); err != nil {
		return CloudFrontFunctionValidationFailed{
			FunctionName: name + ":" + string(stage),
			Failures:     err,
		}
	}
	return nil
}

// TestCloudFrontFunctionE performs a Function test and validates the response.
func TestCloudFrontFunctionE(t testing.TestingT, name string, stage types.FunctionStage, event CloudFrontFunctionEvent) (*CloudFrontTestFunctionResult, error) {
	ctx := context.TODO()

	jsonData, err := json.Marshal(event)
	if err != nil {
		return nil, fmt.Errorf("error serializing CloudFront Function Event: %q", err)
	}

	client := NewCloudFrontclient(t)
	functionDetails, err := client.DescribeFunction(ctx, &cloudfront.DescribeFunctionInput{
		Name:  aws.String(name),
		Stage: stage,
	})
	if err != nil {
		return nil, err
	}

	r, err := client.TestFunction(ctx, &cloudfront.TestFunctionInput{
		EventObject: jsonData,
		IfMatch:     functionDetails.ETag,
		Name:        aws.String(name),
		Stage:       stage,
	})
	if err != nil {
		return nil, err
	}

	return parseTestResult(r.TestResult)
}

// NewCloudFrontclient returns a client for CloudFront. This will fail the test and
// stop execution if there is an error.
func NewCloudFrontclient(t testing.TestingT) *cloudfront.Client {
	sess, err := NewCloudFrontclientE(t)
	require.NoError(t, err)
	return sess
}

// NewCloudFrontclientE returns a client for CloudFront.
func NewCloudFrontclientE(t testing.TestingT) (*cloudfront.Client, error) {
	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion("us-east-1"))
	if err != nil {
		return nil, err
	}
	return cloudfront.NewFromConfig(cfg), nil
}

// assertFunctionStage validates the function stage or fails the test.
func assertFunctionStage(t testing.TestingT, stage string) types.FunctionStage {
	var functionStage types.FunctionStage
	switch stage {
	case string(types.FunctionStageDevelopment):
		functionStage = types.FunctionStageDevelopment
	case string(types.FunctionStageLive):
		functionStage = types.FunctionStageLive
	default:
		t.Errorf("Invalid function stage: %s", stage)
		return types.FunctionStageDevelopment
	}
	return functionStage
}

func parseTestResult(testResult *types.TestResult) (*CloudFrontTestFunctionResult, error) {
	result := CloudFrontTestFunctionResult{
		ExecutionLogs: testResult.FunctionExecutionLogs,
	}
	var err error

	if aws.ToString(testResult.FunctionErrorMessage) != "" {
		result.ErrorMessage = aws.String(invalidFunctionErrorPrefix + *testResult.FunctionErrorMessage)
	}

	if testResult.ComputeUtilization != nil {
		result.Utilization, err = strconv.Atoi(aws.ToString(testResult.ComputeUtilization))
		if err != nil {
			return &result, err
		}
	}
	var output interface{}
	if testResult.FunctionOutput != nil {
		err := json.Unmarshal([]byte(*testResult.FunctionOutput), &output)
		if err != nil {
			return &result, fmt.Errorf("JSON decode error: %w", err)
		}
	}
	result.Output = make(map[string]interface{})
	if output != nil {
		outputMap, ok := output.(map[string]interface{})
		if ok {
			if len(outputMap) != 0 {
				result.Output = outputMap
			}
		}
	}
	return &result, nil
}
