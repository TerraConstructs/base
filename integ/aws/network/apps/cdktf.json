{
  "//": "Dummy file to satisfy the bundling findUp logic"
}
