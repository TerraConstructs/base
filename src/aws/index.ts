// core
// export * from "./tag-aspect";
// export * from "./tag-manager";
export * from "./arn";
export * from "./aws-construct";
export * from "./aws-stack";
export * from "./aws-tags";
export * from "./log-retention";

// core JSII compatible generated interfaces
export * from "./provider-config.generated";

// classes
export * as iam from "./iam";
export * as encryption from "./encryption";
export * as compute from "./compute";
export * as network from "./network";
export * as storage from "./storage";
export * as notify from "./notify";
export * as edge from "./edge";
export * as cloudwatch from "./cloudwatch";
