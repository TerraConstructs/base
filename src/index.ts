export * from "./construct-base";
export * from "./stack-base";
export * from "./duration";
export * from "./size";

// cloud specific exports
export * as aws from "./aws";
