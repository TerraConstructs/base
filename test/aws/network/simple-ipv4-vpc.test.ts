import { Testing } from "cdktf";
import "cdktf/lib/testing/adapters/jest";
import { network, AwsStack } from "../../../src/aws";

const environmentName = "Test";
const gridUUID = "123e4567-e89b-12d3";
const providerConfig = { region: "us-east-1" };
describe("Environment", () => {
  test("Should synth and match SnapShot", () => {
    // GIVEN
    const stack = getAwsStack();
    // WHEN
    new network.SimpleIPv4Vpc(stack, "network", {
      ipv4CidrBlock: "10.0.0.0/16",
      internalDomain: "example.local",
    });
    // THEN
    expect(Testing.synth(stack)).toMatchSnapshot();
  });
  test("Should support adding subnet groups", () => {
    // GIVEN
    const stack = getAwsStack();
    // WHEN
    const vpc = new network.SimpleIPv4Vpc(stack, "network", {
      ipv4CidrBlock: "10.0.0.0/16",
      internalDomain: "example.local",
    });
    vpc.enableDbSubnetGroup();
    vpc.enableElastiCacheSubnetGroup();
    // THEN
    const result = Testing.synth(stack);
    expect(result).toHaveResource({
      tfResourceType: "aws_db_subnet_group",
    });
    expect(result).toHaveResource({
      tfResourceType: "aws_elasticache_subnet_group",
    });
  });
});

function getAwsStack(): AwsStack {
  const app = Testing.app();
  return new AwsStack(app, "TestStack", {
    environmentName,
    gridUUID,
    providerConfig,
  });
}
