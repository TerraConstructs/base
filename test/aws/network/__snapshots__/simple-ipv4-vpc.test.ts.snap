// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment Should synth and match SnapShot 1`] = `
"{
  "data": {
    "aws_availability_zones": {
      "AvailabilityZones": {
        "provider": "aws"
      }
    },
    "aws_caller_identity": {
      "CallerIdentity": {
        "provider": "aws"
      }
    },
    "aws_partition": {
      "Partitition": {
        "provider": "aws"
      }
    }
  },
  "provider": {
    "aws": [
      {
        "region": "us-east-1"
      }
    ]
  },
  "resource": {
    "aws_default_route_table": {
      "network_DefaultRouteTable_7053E46A": {
        "default_route_table_id": "\${aws_vpc.network_39EEAA36.default_route_table_id}",
        "route": [
          {
            "cidr_block": "0.0.0.0/0",
            "core_network_arn": null,
            "destination_prefix_list_id": null,
            "egress_only_gateway_id": null,
            "gateway_id": "\${aws_internet_gateway.network_igw_5DF1EDA1.id}",
            "instance_id": null,
            "ipv6_cidr_block": null,
            "nat_gateway_id": null,
            "network_interface_id": null,
            "transit_gateway_id": null,
            "vpc_endpoint_id": null,
            "vpc_peering_connection_id": null
          }
        ],
        "tags": {
          "Name": "Test-network-default-route-table",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        }
      }
    },
    "aws_eip": {
      "network_PublicSubnet0_Eip_7B2FE34D": {
        "domain": "vpc",
        "tags": {
          "Name": "Test-network-public-subnet-\${element(data.aws_availability_zones.AvailabilityZones.names, 0)}-nat-gateway",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        }
      },
      "network_PublicSubnet1_Eip_47D86D3A": {
        "domain": "vpc",
        "tags": {
          "Name": "Test-network-public-subnet-\${element(data.aws_availability_zones.AvailabilityZones.names, 1)}-nat-gateway",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        }
      }
    },
    "aws_internet_gateway": {
      "network_igw_5DF1EDA1": {
        "tags": {
          "Name": "Test-network-igw",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        },
        "vpc_id": "\${aws_vpc.network_39EEAA36.id}"
      }
    },
    "aws_nat_gateway": {
      "network_PublicSubnet0_NatGateway_CEADEB3D": {
        "allocation_id": "\${aws_eip.network_PublicSubnet0_Eip_7B2FE34D.id}",
        "subnet_id": "\${aws_subnet.network_PublicSubnet0_118EDD7D.id}",
        "tags": {
          "Name": "Test-network-public-subnet-\${element(data.aws_availability_zones.AvailabilityZones.names, 0)}-nat-gateway",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        }
      },
      "network_PublicSubnet1_NatGateway_159BCDA9": {
        "allocation_id": "\${aws_eip.network_PublicSubnet1_Eip_47D86D3A.id}",
        "subnet_id": "\${aws_subnet.network_PublicSubnet1_759DE9E6.id}",
        "tags": {
          "Name": "Test-network-public-subnet-\${element(data.aws_availability_zones.AvailabilityZones.names, 1)}-nat-gateway",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        }
      }
    },
    "aws_route_table": {
      "network_PrivateRouteTable0_28BA0DBE": {
        "route": [
          {
            "carrier_gateway_id": null,
            "cidr_block": "0.0.0.0/0",
            "core_network_arn": null,
            "destination_prefix_list_id": null,
            "egress_only_gateway_id": null,
            "gateway_id": null,
            "ipv6_cidr_block": null,
            "local_gateway_id": null,
            "nat_gateway_id": "\${aws_nat_gateway.network_PublicSubnet0_NatGateway_CEADEB3D.id}",
            "network_interface_id": null,
            "transit_gateway_id": null,
            "vpc_endpoint_id": null,
            "vpc_peering_connection_id": null
          }
        ],
        "tags": {
          "Name": "Test-network-private-route-table-\${element(data.aws_availability_zones.AvailabilityZones.names, 0)}",
          "aws-cdk:subnet-name": "Public",
          "aws-cdk:subnet-type": "Public",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3",
          "kubernetes.io/role/elb": "1"
        },
        "vpc_id": "\${aws_vpc.network_39EEAA36.id}"
      },
      "network_PrivateRouteTable1_163E06E8": {
        "route": [
          {
            "carrier_gateway_id": null,
            "cidr_block": "0.0.0.0/0",
            "core_network_arn": null,
            "destination_prefix_list_id": null,
            "egress_only_gateway_id": null,
            "gateway_id": null,
            "ipv6_cidr_block": null,
            "local_gateway_id": null,
            "nat_gateway_id": "\${aws_nat_gateway.network_PublicSubnet1_NatGateway_159BCDA9.id}",
            "network_interface_id": null,
            "transit_gateway_id": null,
            "vpc_endpoint_id": null,
            "vpc_peering_connection_id": null
          }
        ],
        "tags": {
          "Name": "Test-network-private-route-table-\${element(data.aws_availability_zones.AvailabilityZones.names, 1)}",
          "aws-cdk:subnet-name": "Public",
          "aws-cdk:subnet-type": "Public",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3",
          "kubernetes.io/role/elb": "1"
        },
        "vpc_id": "\${aws_vpc.network_39EEAA36.id}"
      }
    },
    "aws_route_table_association": {
      "network_DataSubnet0_RouteTableAssociation_25025986": {
        "route_table_id": "\${aws_route_table.network_PrivateRouteTable0_28BA0DBE.id}",
        "subnet_id": "\${aws_subnet.network_DataSubnet0_26F830AE.id}"
      },
      "network_DataSubnet1_RouteTableAssociation_23B57B58": {
        "route_table_id": "\${aws_route_table.network_PrivateRouteTable1_163E06E8.id}",
        "subnet_id": "\${aws_subnet.network_DataSubnet1_0A85069C.id}"
      },
      "network_PrivateSubnet0_RouteTableAssociation_56D48E4C": {
        "route_table_id": "\${aws_route_table.network_PrivateRouteTable0_28BA0DBE.id}",
        "subnet_id": "\${aws_subnet.network_PrivateSubnet0_4BC47064.id}"
      },
      "network_PrivateSubnet1_RouteTableAssociation_8A034792": {
        "route_table_id": "\${aws_route_table.network_PrivateRouteTable1_163E06E8.id}",
        "subnet_id": "\${aws_subnet.network_PrivateSubnet1_EEFFD3F4.id}"
      }
    },
    "aws_service_discovery_private_dns_namespace": {
      "network_ServiceDiscoveryNamespace_B9CE35BD": {
        "description": "Private DNS namespace for Test-network",
        "name": "example.local",
        "tags": {
          "Name": "Test-network-service-discovery-namespace",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        },
        "vpc": "\${aws_vpc.network_39EEAA36.id}"
      }
    },
    "aws_subnet": {
      "network_DataSubnet0_26F830AE": {
        "availability_zone": "\${element(data.aws_availability_zones.AvailabilityZones.names, 0)}",
        "cidr_block": "\${cidrsubnet(cidrsubnet(\\"10.0.0.0/16\\", 2, 2), 2, 0)}",
        "map_public_ip_on_launch": false,
        "tags": {
          "Name": "Test-network-data-subnet-\${element(data.aws_availability_zones.AvailabilityZones.names, 0)}",
          "aws-cdk:subnet-name": "Data",
          "aws-cdk:subnet-type": "Isolated",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        },
        "vpc_id": "\${aws_vpc.network_39EEAA36.id}"
      },
      "network_DataSubnet1_0A85069C": {
        "availability_zone": "\${element(data.aws_availability_zones.AvailabilityZones.names, 1)}",
        "cidr_block": "\${cidrsubnet(cidrsubnet(\\"10.0.0.0/16\\", 2, 2), 2, 1)}",
        "map_public_ip_on_launch": false,
        "tags": {
          "Name": "Test-network-data-subnet-\${element(data.aws_availability_zones.AvailabilityZones.names, 1)}",
          "aws-cdk:subnet-name": "Data",
          "aws-cdk:subnet-type": "Isolated",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        },
        "vpc_id": "\${aws_vpc.network_39EEAA36.id}"
      },
      "network_PrivateSubnet0_4BC47064": {
        "availability_zone": "\${element(data.aws_availability_zones.AvailabilityZones.names, 0)}",
        "cidr_block": "\${cidrsubnet(cidrsubnet(\\"10.0.0.0/16\\", 2, 1), 2, 0)}",
        "map_public_ip_on_launch": false,
        "tags": {
          "Name": "Test-network-private-subnet-\${element(data.aws_availability_zones.AvailabilityZones.names, 0)}",
          "aws-cdk:subnet-name": "Private",
          "aws-cdk:subnet-type": "Private",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3",
          "kubernetes.io/role/internal-elb": "1"
        },
        "vpc_id": "\${aws_vpc.network_39EEAA36.id}"
      },
      "network_PrivateSubnet1_EEFFD3F4": {
        "availability_zone": "\${element(data.aws_availability_zones.AvailabilityZones.names, 1)}",
        "cidr_block": "\${cidrsubnet(cidrsubnet(\\"10.0.0.0/16\\", 2, 1), 2, 1)}",
        "map_public_ip_on_launch": false,
        "tags": {
          "Name": "Test-network-private-subnet-\${element(data.aws_availability_zones.AvailabilityZones.names, 1)}",
          "aws-cdk:subnet-name": "Private",
          "aws-cdk:subnet-type": "Private",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3",
          "kubernetes.io/role/internal-elb": "1"
        },
        "vpc_id": "\${aws_vpc.network_39EEAA36.id}"
      },
      "network_PublicSubnet0_118EDD7D": {
        "availability_zone": "\${element(data.aws_availability_zones.AvailabilityZones.names, 0)}",
        "cidr_block": "\${cidrsubnet(cidrsubnet(\\"10.0.0.0/16\\", 2, 0), 2, 0)}",
        "map_public_ip_on_launch": true,
        "tags": {
          "Name": "Test-network-public-subnet-\${element(data.aws_availability_zones.AvailabilityZones.names, 0)}",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        },
        "vpc_id": "\${aws_vpc.network_39EEAA36.id}"
      },
      "network_PublicSubnet1_759DE9E6": {
        "availability_zone": "\${element(data.aws_availability_zones.AvailabilityZones.names, 1)}",
        "cidr_block": "\${cidrsubnet(cidrsubnet(\\"10.0.0.0/16\\", 2, 0), 2, 1)}",
        "map_public_ip_on_launch": true,
        "tags": {
          "Name": "Test-network-public-subnet-\${element(data.aws_availability_zones.AvailabilityZones.names, 1)}",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        },
        "vpc_id": "\${aws_vpc.network_39EEAA36.id}"
      }
    },
    "aws_vpc": {
      "network_39EEAA36": {
        "cidr_block": "10.0.0.0/16",
        "enable_dns_hostnames": true,
        "enable_dns_support": true,
        "tags": {
          "Name": "Test-network",
          "grid:EnvironmentName": "Test",
          "grid:UUID": "123e4567-e89b-12d3"
        }
      }
    }
  },
  "terraform": {
    "required_providers": {
      "aws": {
        "source": "aws",
        "version": "5.100.0"
      }
    }
  }
}"
`;
